index.d.ts

```diff
@@ -115,16 +118,30 @@ export namespace Cmd {
     args?: any[]
   ): MapCmd<A>;
 
-  export function run<A extends Action, B extends Action>(
-    f: Function,
+  // Allow the use of special dispatch | getState symbols
+  type ArgOrSymbol<T> = {
+    [K in keyof T]: T[K] extends GetState
+      ? typeof getState
+      : T[K] extends Dispatch
+        ? typeof dispatch
+        : T[K];
+  }
+  export type PromiseResult<T> = T extends Promise<infer U> ? U : T;
+
+  export function run<
+    Func extends (...args: any[]) => Promise<any> | any,
+    SuccessAction extends Action,
+    FailAction extends Action,
+    >(
+    f: Func,
     options?: {
-      args?: any[];
-      failActionCreator?: ActionCreator<A>;
-      successActionCreator?: ActionCreator<B>;
+      args?: ArgOrSymbol<Parameters<Func>>;
```

Comment:
I wonder if we could use `infer` to actually determine which arguments need to be provided to the function so that we can get type checking.

Reply:
Can you explain what you mean? `Parameters` is a typescript built-in which uses infer internally.

Reply:
Ah, I didn't realize that. I think this is already doing what I was hoping it was, haha. Disregard.